File: Pemrograman-1/Modul 6/C/PRAK604-2310817210014-DamarjatiSuryoLaksono.c
#include <stdio.h>
#include <string.h>

int main() {
    char code[100], message[100];
    fgets(code, sizeof(code), stdin);
    code[strcspn(code, "\n")] = '\0';
    fgets(message, sizeof(message), stdin);
    message[strcspn(message, "\n")] = '\0';
    if (strlen(code) != strlen(message)) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
        return 0;
    }
    int starCount = 0, hashCount = 0;
    for (int i = 0; i < strlen(message); i++) {
        if (message[i] == ' ' && code[i] == ' ' ) {
            continue;
        } if (message[i] == code[i]) {
            message[i] = '*';
            starCount++;
        } else {
            message[i] = '#';
            hashCount++;
        }
    }
    printf("%s\n", message);
    printf("* = %d\n", starCount);
    printf("# = %d\n", hashCount);
    if (starCount >= hashCount) {
        printf("Pesan Asli\n");
    } else {
        printf("Pesan Palsu\n");
    }

    return 0;
}

File: Pemrograman-1/Modul 6/Python/PRAK604-2310817210014-DamarjatiSuryoLaksono.py
code = input().rstrip('\n')
message = input().rstrip('\n')

if len(code) != len(message):
    print("Panjang kalimat berbeda, pesan palsu")
    exit()

star_count = 0
hash_count = 0
modified_message = ""

for i in range(len(message)):
    if message[i] == ' ' and code[i] == ' ':
        continue
    elif message[i] == code[i]:
        modified_message += '*'
        star_count += 1
    else:
        modified_message += '#'
        hash_count += 1

print(modified_message)
print("* =", star_count)
print("# =", hash_count)

if star_count >= hash_count:
    print("Pesan Asli")
else:
    print("Pesan Palsu")
